killerbee/kbutils.py

```diff
@@ -126,19 +126,22 @@ def is_valid_channel(self, channel, page= 0):
         Based on sniffer capabilities, return if this is an OK channel number.
         @rtype: Boolean
         '''
-        # if sub-ghz, check that page and channel match
+        # if sub-ghz, check that page and channel and capability match
         if page:
-            if (page == 28 or page == 30 or page == 31) and channel > 26:
+            if page == 28 and (channel > 26 or not self.check(self.FREQ_863)):
                 return False
-            elif page == 29 and channel  > 8:
+            elif page == 29 and (channel > 8 or not self.check(self.FREQ_868)):
```

Comment:
You are switching between `if` and `elif` in this chain. It doesn't effect code flow at all, but might worth being consistent.

Reply:
Yeah, that's what happens when you squeeze your own code in amongst someone else's! :) I prefer the raw 'if' myself... Thanks again for the spot.